pkg/collector/reconcile/deployment_test.go

```diff
@@ -55,6 +116,54 @@ func TestExpectedDeployments(t *testing.T) {
 		assert.Equal(t, int32(2), *actual.Spec.Replicas)
 	})
 
+	t.Run("should not update target allocator deployment when the config is updated", func(t *testing.T) {
```

Comment:
Could you please also change, say, the number of replicas? This is to demonstrate that you thought about the case where users are changing the deployment manually and you are OK with that.